### PR TITLE
[14.0][IMP] subscription_oca: more invoicing modes

### PR DESCRIPTION
**Pull request by ilyasProgrammer**
New modes to generate SO and Invoce:

            ("sale_and_invoice_draft", "Sale order & Invoice Draft"),
            ("sale_and_invoice_send", "Sale Order & Invoice Send"),
            ("sale_draft", "Sale Order Draft"),
            ("sale_confirmed", "Sale Order Confirmed"),

### CODE REVIEW

File: subscription_oca/models/sale_subscription_template.py [hunk 1]
```diff
@@ -35,6 +35,10 @@ class SaleSubscriptionTemplate(models.Model):
             ("invoice", "Invoice"),
             ("invoice_send", "Invoice & send"),
             ("sale_and_invoice", "Sale order & Invoice"),
+            ("sale_and_invoice_draft", "Sale order & Invoice Draft"),
+            ("sale_and_invoice_send", "Sale Order & Invoice Send"),
+            ("sale_daft", "Sale Order Draft"),
```
Comment (chrisandrewmann) [inline subscription_oca/models/sale_subscription_template.py]: "sale_daft" should be "sale_draft"

File: subscription_oca/models/sale_subscription.py [hunk 2]
```diff
@@ -321,35 +321,52 @@ def generate_invoice(self):
...
-                    )
+                    self.send_invoice(invoice)
                 invoice_number = invoice.name
                 message_body = (
                     "<b>%s</b> <a href=# data-oe-model=account.move data-oe-id=%d>%s</a>"
                     % (msg_static, invoice.id, invoice_number)
                 )
 
-        if self.template_id.invoicing_mode == "sale_and_invoice":
+        elif self.template_id.invoicing_mode in ["sale_daft"]:
```
Comment (chrisandrewmann) [inline subscription_oca/models/sale_subscription.py]: ["sale_daft"] should be ["sale_draft"]

**Review by francesco-ooops — APPROVED** [review]
Functional ok!

**Review by HekkiMelody — APPROVED** [review]
Code review, LGTM

**Pull request merged**